sound/soc/sof/sof-priv.h

```diff
@@ -67,6 +67,9 @@ struct snd_sof_pdata;
  * and DSP device(s).
  */
 struct snd_sof_dsp_ops {
+
+	/* The following ops are mandatory */
+
 	/* probe and remove */
 	int (*remove)(struct snd_sof_dev *sof_dev);
```

Comment:
why should .remove be mandatory? if all your driver uses devm_ stuff and the hardware is reset correctly in the .probe, why would we care?

Reply:
@plbossart going by what remove() does for SKL+, we also disable the DSP in remove(). Given that we are not really unloading the topology from the DSP like we do in the case of load, shouldnt we also be turning off the DSP here?

Reply:
@plbossart i believe for module load/unload to work in SOF, we should make remove() mandatory so that the DSP can be turned OFF here. Please let me know if you disagree and I'll make changes accordingly.